enumfields/fields.py

```diff
@@ -34,7 +36,11 @@ def to_python(self, value):
         raise ValidationError('%s is not a valid value for enum %s' % (value, self.enum), code="invalid_enum_value")
 
     def get_prep_value(self, value):
-        return None if value is None else self.enum(value).value
+        if value is None:
+            return None
+        if not isinstance(value, self.enum):
+            return self.enum(value).value
+        return value.value  # Already the correct type
```

Comment:
Hmm.. sounds like there could just as well be `assert isinstance`.

As said in the documentation of get_prep_value [1]:
> value is the current value of the model’s attribute, and the method should return data in a format that has been prepared for use as a parameter in a query.

So how could model's attribute be something else than None or instance of the enum? I don't think it can.

[1] https://docs.djangoproject.com/en/1.9/ref/models/fields/#django.db.models.Field.get_prep_value